x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -367,29 +374,34 @@ func createMetricDataQueries(listMetricsTotal []metricsWithStatistics, dataGranu
 			dataGranularityInSec := int32(dataGranularity.Seconds())
 
 			id := "cw" + strconv.Itoa(i) + "stats" + strconv.Itoa(j)
-			metricDataQueries = append(metricDataQueries, types.MetricDataQuery{
+			metricDataQuery := types.MetricDataQuery{
 				Id: &id,
 				MetricStat: &types.MetricStat{
 					Period: &dataGranularityInSec,
 					Stat:   &stat,
-					Metric: &metric,
+					Metric: &metric.Metric,
 				},
 				Label: &label,
-			})
+			}
+			if listMetric.cloudwatchMetric.AccountID != "" {
+				metricDataQuery.AccountId = &metric.AccountID
+			}
+
+			metricDataQueries = append(metricDataQueries, metricDataQuery)
 		}
 	}
 	return metricDataQueries
 }
 
-func constructLabel(metric types.Metric, statistic string) string {
-	// label = metricName + namespace + statistic + dimKeys + dimValues
-	label := *metric.MetricName + labelSeparator + *metric.Namespace + labelSeparator + statistic
+func constructLabel(metric aws.MetricWithID, statistic string) string {
+	// label = accountID + accountLabel + metricName + namespace + statistic + dimKeys + dimValues
+	label := metric.AccountID + labelSeparator + "${PROP('AccountLabel')}" + labelSeparator + *metric.Metric.MetricName + labelSeparator + *metric.Metric.Namespace + labelSeparator + statistic
```

Comment:
NIT: I think all the label definitions would look cleaner with something like `strings.Join(<array of label components>, labelSeparator)`